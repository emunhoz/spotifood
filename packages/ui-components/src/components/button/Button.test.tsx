import React from 'react'
import { render, fireEvent, screen } from '../../utils/test-utils'
import { Button } from './Button'

describe('[Component]: Button', () => {
  const mockOnClickFunction = jest.fn()

  test('render the component', () => {
    const { asFragment } = render(<Button>Hello world</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('render the component', () => {
    const { asFragment } = render(<Button ghost>Hello world</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('render the component', () => {
    const { asFragment } = render(<Button negative>Hello world</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('should trigger function when button is pressed', () => {
    render(<Button onClick={mockOnClickFunction}>Hello world</Button>)

    fireEvent.click(screen.getByText('Hello world'))

    expect(mockOnClickFunction).toHaveBeenCalled()
  })
})
