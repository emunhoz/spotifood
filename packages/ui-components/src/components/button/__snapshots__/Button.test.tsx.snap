// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Button render component disabled 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ fIJaZi"
    disabled=""
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render component with big size style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ hJvuMX"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render component with small size style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ fdxWyO"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render ghost style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ ljIJrW"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render negative style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ bMkSsV"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render the component 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ fIJaZi"
  >
    Hello world
  </button>
</DocumentFragment>
`;
